### PR TITLE
Security fixes, close open steams

### PR DESCRIPTION
**Pull request by tomas-sexenian**
Issue: 103465

First wave of fixes for scanning alerts 190, 189, 188, 187, 186, 185, 184, 182, 178, 176, 175, 174, 173, 172, 171, 170, 169 and 168

### CODE REVIEW

File: gxmail/src/main/java/com/genexus/internet/POP3SessionJavaMail.java [hunk 1]
```diff
@@ -316,7 +316,9 @@
 			gxmessage.setHtmltext(newHTML);
 		}
 
-		saveFile(fileName, part.getContent() instanceof MimeMessage ? ((MimeMessage) part.getContent()).getInputStream() : part.getInputStream());
+		try (InputStream is = part.getContent() instanceof MimeMessage ? ((MimeMessage) part.getContent()).getInputStream() : part.getInputStream()){
```
[the remark this answers is not in the document]
Comment (sgrampone) [reply gxmail/src/main/java/com/genexus/internet/POP3SessionJavaMail.java:319]: False positive, it is inside a try, it closes automatically.

[the remark this answers is not in the document]
Comment (sgrampone) [reply gxmail/src/main/java/com/genexus/internet/POP3SessionJavaMail.java:319]: False positive, it is inside a try, it closes automatically.

File: common/src/main/java/com/genexus/Messages.java [hunk 2]
```diff
@@ -114,14 +114,13 @@
...
             {
-              BufferedReader bufread = new BufferedReader(new InputStreamReader(is, "UTF8"));
-              line = bufread.readLine();
-              while (line != null)
-              {
-                parseLine(line);
-                line = bufread.readLine();
-              }
-              bufread.close();
+              try (BufferedReader bufread = new BufferedReader(new InputStreamReader(is, "UTF8"))) {
```
[the remark this answers is not in the document]
Comment (sgrampone) [reply common/src/main/java/com/genexus/Messages.java:117]: False positive, the stream is closed a couple of lines after.

File: common/src/main/java/com/genexus/ImagesPath.java [hunk 3]
```diff
@@ -89,14 +89,13 @@
 				if (is != null)
 				{
 					parseLocations = false;
-					BufferedReader bufread = new BufferedReader(new InputStreamReader(is, "UTF8"));
-					line = bufread.readLine();
-					while (line != null)
-					{
-						parseLine(line, KBId);
+					try (BufferedReader bufread = new BufferedReader(new InputStreamReader(is, "UTF8"))) {
```
[the remark this answers is not in the document]
Comment (sgrampone) [reply common/src/main/java/com/genexus/ImagesPath.java:92]: False positive, the stream is closed a couple of lines after.

File: common/src/main/java/com/genexus/URLRouter.java [hunk 4]
```diff
@@ -137,14 +137,13 @@
 
 					if (is != null)
 					{
-						BufferedReader bufread = new BufferedReader(new InputStreamReader(is, "UTF8"));
-						line = bufread.readLine();
-						while (line != null)
-						{
-							parseLine(line);
+						try (BufferedReader bufread = new BufferedReader(new InputStreamReader(is, "UTF8"))) {
```
[the remark this answers is not in the document]
Comment (sgrampone) [reply common/src/main/java/com/genexus/URLRouter.java:140]: False positive, the stream is closed a couple of lines after.

**Review by sgrampone — DISMISSED** [review]

**Comment by tomas-sexenian** [comment]
Targeted security alerts: 401, 400, 208, 207, 195 ,182 ,160 ,357

**Review by sgrampone — APPROVED** [review]

File: java/src/main/java/com/genexus/util/GXFile.java [hunk 5]
```diff
@@ -787,16 +787,17 @@
                 setUnknownError(e);
             }
         }
-        if (lineIterator != null) {
-           	try {
+		try {
+			if (this.getStream() != null)
```
[the remark this answers is not in the document]
Comment (sgrampone) [reply java/src/main/java/com/genexus/util/GXFile.java:791]: The stream is closed on finally block.

**Review by sgrampone — APPROVED** [review]

**Pull request merged**